src/SFML/Graphics/Font.cpp

```diff
@@ -840,8 +841,9 @@ bool Font::setCurrentSize(unsigned int characterSize) const
 Font::Page::Page(bool smooth) : nextRow(3)
 {
     // Make sure that the texture is initialized by default
-    sf::Image image;
-    image.create({128, 128}, Color(255, 255, 255, 0));
+    std::optional<Image> maybeImage = Image::create({128, 128}, Color(255, 255, 255, 0));
+    assert(maybeImage.has_value());
+    Image& image = *maybeImage;
```

Comment:
This one is simpler and more efficient as it does less operations -- it just takes a reference to the `Image` stored inside the `optional`'s buffer.

Moving it out requires a move constructor call, and also makes reusing the `optional` undefined behavior, as it's now in a "valid but unspecified" state.

This should be the technique we use unless we explicitly intend to take ownership from the `optional`.

Reply:
> This should be the technique we use unless we explicitly intend to take ownership from the optional.

This is part of what I was curious about. Stealing ownership seems ideal because it signals that the optional has done all the work it's ever going to do in this scope and should never be reused. Is it worth leaving that optional in a UB-prone state though? I'm unsure.

Reply:
We could do something like this:
```cpp
template <typename T>
T unwrap(std::optional<T>&& movedFrom) // or by-value?
{
    assert(movedFrom.has_value());
    *movedFrom // does this really move out?
}
```
I'm probably getting all the references and moves wrong, we would need to make sure the destructor is not invoked. 
But the idea is that we could directly call this on temporaries:
```cpp
Image image = unwrap(Image::create(...));
```
This avoids the need for a disengaged `maybeImage` in scope, and is also shorter.

Reply:
I.e. similar to `.value()`, but with an assertion instead of exception.

Reply:
I think an abstraction like this ought to leave the optional empty instead of containing a moved-from instances of `T`.

Reply:
I'm experimenting with this idea and I think it makes sense for the parameter to be an `optional<T>&` so that this only works with lvalues. That's when we'd actually want this function. For rvalues you can just use either of the two options for extracting the value out of an optional.
```cpp
template <typename T>
[[nodiscard]] constexpr auto unwrap(std::optional<T>& optional) {
    assert(optional.has_value());
    auto value = std::move(*optional);
    optional.reset();
    return value;
}
```
This will call the destructor on a moved-from instances of T but I'm okay with that for the added safety of nulling out the optional after we're done with it.